pkg/util/leaderelection.go

```diff
@@ -10,22 +10,26 @@ import (
 	"k8s.io/klog"
 )
 
-func GetLeaderElectionConfig(ctx context.Context, restcfg *rest.Config) configv1.LeaderElection {
+// GetLeaderElectionConfig returns leader election configs defaults based on the cluster topology
+func GetLeaderElectionConfig(restcfg *rest.Config, enabled bool) configv1.LeaderElection {
```

Comment:
Why do you need this ultis for leader election? 
By default, you have the C+R implementation for leader election : https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/leaderelection/leader_election.go

What is the extra requirement on top of that? 

Also, you might give a look in the doc https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#leader-election which has compressive info about it.